File: split/printchar.pas
procedure printchar(s: ASCIIcode);
label
  10;
begin
  if{244:}
s = eqtb[5312].int{:244}
 then
    if selector < 20 then
    begin
      println;
      goto 10;
    end;
  case selector of
    19:
    begin
      write(termout, xchr[s]);
      write(logfile, xchr[s]);
      termoffset := termoffset + 1;
      fileoffset := fileoffset + 1;
      if termoffset = maxprintline then
      begin
        writeln(termout);
        termoffset := 0;
      end;
      if fileoffset = maxprintline then
      begin
        writeln(logfile);
        fileoffset := 0;
      end;
    end;
    18:
    begin
      write(logfile, xchr[s]);
      fileoffset := fileoffset + 1;
      if fileoffset = maxprintline then
        println;
    end;
    17:
    begin
      write(termout, xchr[s]);
      termoffset := termoffset + 1;
      if termoffset = maxprintline then
        println;
    end;
    16: ;
    20: if tally < trickcount then
        trickbuf[tally mod errorline] := s;
    21: if poolptr < poolsize then
      begin
        strpool[poolptr] := s;
        poolptr := poolptr + 1;
      end;
    others: write(writefile[selector], xchr[s])
  end;
  tally := tally + 1;
  10: ;
end;{:58}
{59:}

File: cpp/printchar.cpp
#include "printchar.h"
#include "println.h"
#include <iostream>

void printchar(ASCIIcode s)
{
	if (s == new_line_char())
		if (selector < pseudo)
		{
			println();
			return;
		}
	switch (selector)
	{
		case term_and_log:
			std::cout << xchr[s];
			logfile << xchr[s];
			termoffset++;
			fileoffset++;
			if (termoffset == maxprintline)
			{
				std::cout << "\n";
				termoffset = 0;
			}
			if (fileoffset == maxprintline)
			{
				logfile << "\n";
				fileoffset = 0;
			}
			break;
		case log_only:
			logfile << xchr[s];
			fileoffset++;
			if (fileoffset == maxprintline)
			println();
			break;
		case term_only:
			std::cout << xchr[s];
			termoffset++;
			if (termoffset == maxprintline)
				println();
			break;
		case no_print:
			break;
		case pseudo: 
			if (tally < trickcount)
				trickbuf[tally%errorline] = s;
			break;
		case new_string: 
			if (poolptr < poolsize)
			{
				strpool[poolptr] = s;
				poolptr++;
			}
			break;
		default: 
			writefile[selector] << xchr[s];
	}
	tally++;
}
